### PR TITLE
Allen/new bridge

### PR DESCRIPTION
**Pull request by ahujsak**
This pull request sets up the new yaob bridge to inject into web views for partners.
Asana : Implement edgeProvider plugin object API
https://app.asana.com/0/361770107085503/1001052953443732

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

### CODE REVIEW

File: src/components/scenes/PluginViewScene.js [hunk 1]
```diff
@@ -271,6 +344,8 @@ class PluginView extends React.Component<PluginProps, PluginState> {
                 this.openingSendConfirmation = false
               }
             }
+            console.log('a1: info', info)
```
Comment (paullinator) [inline src/components/scenes/PluginViewScene.js]: Extraneous debug logging. Please remove for production

File: src/components/scenes/PluginViewScene.js [hunk 2]
```diff
@@ -252,6 +324,7 @@ class PluginView extends React.Component<PluginProps, PluginState> {
           return
         }
         this.openingSendConfirmation = true
+        console.log('a1: parsed uri ', data['edge-uri'])
```
Comment (paullinator) [inline src/components/scenes/PluginViewScene.js]: Extraneous debug logging. Please remove for production

File: src/components/scenes/PluginViewScene.js [hunk 3]
```diff
@@ -304,14 +379,15 @@ class PluginView extends React.Component<PluginProps, PluginState> {
           if (this.openingSendConfirmation) {
             return
           }
+
           this.openingSendConfirmation = true
+          // console.log('a1: data ', data)
```
Comment (paullinator) [inline src/components/scenes/PluginViewScene.js]: Extraneous debug logging. Please remove for production

File: src/modules/UI/scenes/Plugins/bridgeApi.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    return Promise.resolve(wallet.receiveAddress)
+  }
+  // Write data to user's account. This data is encrypted and persisted in their Edge
+  // account and transferred between devices
+  async writeData (data: { [key: string]: string }) {
+    const account = CORE_SELECTORS.getAccount(this._state)
+    const folder = account.pluginData
+    try {
+      for (const key in data) {
+        await folder.setItem(this._plugin, key, data[key])
```
Comment (paullinator) [inline src/modules/UI/scenes/Plugins/bridgeApi.js]: This should use Promise.all()

File: src/components/scenes/PluginViewScene.js (hunk 1 above)
Comment (ahujsak) [reply src/components/scenes/PluginViewScene.js]: fixed

File: src/components/scenes/PluginViewScene.js (hunk 2 above)
Comment (ahujsak) [reply src/components/scenes/PluginViewScene.js]: fixed

File: src/components/scenes/PluginViewScene.js (hunk 3 above)
Comment (ahujsak) [reply src/components/scenes/PluginViewScene.js]: fixed

File: src/modules/UI/scenes/Plugins/bridgeApi.js (hunk 4 above)
Comment (swansontec) [reply src/modules/UI/scenes/Plugins/bridgeApi.js]:
```
await Promise.all(Object.keys(data, key => folder.setItem(plugin, key, data[key])))
```

File: .eslintignore [hunk 5]
```diff
@@ -1,2 +1,5 @@
 node_modules
 **/scratch.*
+src/lib/bridge/rolledUp.js
+src/lib/bridge/bridge.js
+src/lib/bridge/injectThisInWebView.js
```
Comment (swansontec) [inline .eslintignore]: Add to .gitignore?

File: src/lib/bridge/bridge.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Bridge } from 'yaob'
```
Comment (swansontec) [inline src/lib/bridge/bridge.js]: /* global window */
// eslint-ignore no-flow-whatever

File: src/lib/bridge/injectThisInWebView.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+export const javascript =
```
Comment (swansontec) [inline src/lib/bridge/injectThisInWebView.js]: Remove from git.

File: src/components/scenes/PluginViewScene.js [hunk 8]
```diff
@@ -132,6 +183,9 @@ class PluginView extends React.Component<PluginProps, PluginState> {
     this.plugin = this.props.plugin
     this.plugin.environment.apiKey = ENV.PLUGIN_API_KEYS ? ENV.PLUGIN_API_KEYS[this.plugin.name] : 'edgeWallet' // latter is dummy code
     this.updateBridge(this.props)
+    this.yaobBridge = new Bridge({
```
Comment (swansontec) [inline src/components/scenes/PluginViewScene.js]: Move to `webviewLoaded`.

File: src/modules/UI/scenes/Plugins/bridgeApi.js [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+  // This overrides any parameters specified in a URI
+  uniqueIdentifier?: string
+}
+
+type EdgeGetReceiveAddressOptions = {
+  // Metadata to tag these addresses with for when funds arrive at the address
+  metadata?: EdgeMetadata
+}
+
+class EdgeProvider {
```
Comment (swansontec) [inline src/modules/UI/scenes/Plugins/bridgeApi.js]: `extends Bridgeable`

File: src/components/scenes/PluginViewScene.js [hunk 10]
```diff
@@ -349,6 +418,11 @@ class PluginView extends React.Component<PluginProps, PluginState> {
     }
   }
 
+  webviewLoaded = () => {
+    const edgeProvider = new EdgeProvider(this.props.plugin, this.props.currentState, this.props.thisDispatch)
+    bridgifyObject(edgeProvider)
```
Comment (swansontec) [inline src/components/scenes/PluginViewScene.js]: Not needed if `class EdgeProvider extends Bridgeable`

File: src/components/scenes/PluginViewScene.js [hunk 11]
```diff
@@ -211,6 +265,18 @@ class PluginView extends React.Component<PluginProps, PluginState> {
   _nextMessage = datastr => {
     this.webview.injectJavaScript(`window.PLUGIN_NEXT('${datastr}')`)
   }
+  bridgeMessageProcessing = (data: any) => {
+    this.yaobBridge.handleMessage(data)
```
Comment (swansontec) [inline src/components/scenes/PluginViewScene.js]: Just inline this into _onMessage.

**Comment by paullinator** [comment]
Since we're not in a hurry to put this change in, @ahujsak can you implement @swansontec 's requested changes before we merge.

File: src/modules/UI/scenes/Plugins/bridgeApi.js (hunk 4 above)
Comment (ahujsak) [reply src/modules/UI/scenes/Plugins/bridgeApi.js]: Done

File: src/lib/bridge/bridge.js (hunk 6 above)
Comment (ahujsak) [reply src/lib/bridge/bridge.js]: done

File: src/modules/UI/scenes/Plugins/bridgeApi.js (hunk 9 above)
Comment (ahujsak) [reply src/modules/UI/scenes/Plugins/bridgeApi.js]: done

File: src/components/scenes/PluginViewScene.js (hunk 11 above)
Comment (ahujsak) [reply src/components/scenes/PluginViewScene.js]: done

File: .eslintignore (hunk 5 above)
Comment (ahujsak) [reply .eslintignore]: done

File: src/components/scenes/PluginViewScene.js (hunk 8 above)
Comment (ahujsak) [reply src/components/scenes/PluginViewScene.js]: done

File: src/components/scenes/PluginViewScene.js (hunk 10 above)
Comment (ahujsak) [reply src/components/scenes/PluginViewScene.js]: done

**Pull request merged**